src/shared/actions/flags.ts

```diff
@@ -62,3 +73,19 @@ export const getFlags = () => async (
     dispatch(setFlags(RemoteDataState.Error, null))
   }
 }
+
+export const getPublicFlags = () => async (dispatch: Dispatch<Actions>) => {
+  try {
+    const url = `${getAPIBasepath()}/api/v2private/flags`
+    const response = await fetch(url)
+    const flags = await response.json()
+
+    if (flags?.status && flags?.status !== 200) {
+      throw new Error(flags.message)
+    }
+
+    dispatch(setPublicFlags(flags))
+  } catch (error) {
+    console.error(error)
+  }
```

Comment:
if you're not going to do anything with the error, let it fail up to the error boundary. As it is now, this swallows errors and makes us unaware of them unless we happen to be the client that generates one and sees it in the console.

https://digitaldrummerj.me/dont-swallow-the-exceptions/